x-pack/legacy/plugins/ml/public/application/data_frame_analytics/common/analytics.ts

```diff
@@ -341,3 +394,53 @@ export const loadEvalData = async ({
     return results;
   }
 };
+
+interface TrackTotalHitsSearchResponse {
+  hits: {
+    total: {
+      value: number;
+      relation: string;
+    };
+    hits: any[];
+  };
+}
+
+export const loadDocsCount = async ({
+  ignoreDefaultQuery = true,
+  isTraining,
+  searchQuery,
+  resultsField,
+  destIndex,
+}: {
+  ignoreDefaultQuery?: boolean;
+  isTraining: boolean;
+  searchQuery: SavedSearchQuery;
+  resultsField: string;
+  destIndex: string;
+}): Promise<{
+  docsCount: number | null;
+  success: boolean;
+}> => {
```

Comment:
Suggest to move to interfaces e.g. `: LoadDocsCountConfig` and `Promise<LoadDocsCountResponse>`.